substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoFeature.java

```diff
@@ -85,6 +85,22 @@ public void afterRegistration(AfterRegistrationAccess access) {
         }
     }
 
+    @Override
+    public void beforeAnalysis(BeforeAnalysisAccess access) {
+        /*
+         * Make the name provider aware of the native libs
+         */
+        if (!UniqueShortNameProviderDefaultImpl.UseDefault.useDefaultProvider()) {
+            if (ImageSingletons.contains(UniqueShortNameProvider.class)) {
+                UniqueShortNameProvider provider = ImageSingletons.lookup(UniqueShortNameProvider.class);
+                if (provider instanceof NativeImageBFDNameProvider) {
+                    var accessImpl = (FeatureImpl.BeforeAnalysisAccessImpl) access;
```

Comment:
Instead of this complex chain of checks to determine if you have a `NativeImageBFDNameProvider` instance just add a private field to class `NativeImageDebugInfoFeature` that holds a ref to the `NativeImageBFDNameProvider` you instantiated in `afterRegistration`. I.e. in `afterRegistration` you change:
```java
ImageSingletons.add(UniqueShortNameProvider.class, new NativeImageBFDNameProvider(ignored));
``` to
```java
bfdNameProvider = new NativeImageBFDNameProvider(ignored);
ImageSingletons.add(UniqueShortNameProvider.class, bfdNameProvider);
```
Then here you can just use
```java
if (bfdNameProvider != null) { ... }
```

Reply:
I just pushed a fix for this